### PR TITLE
Restore bracket on zio actions

### PR DESCRIPTION
**Pull request by regiskuckaertz**
The method existed only inside an implicit class for invariant types, but it doesn't seem necessary. Besides it's missing from the scaladoc which is quite surprising (since `bracket_` is there).

### CODE REVIEW

File: core/shared/src/main/scala/zio/ZIO.scala [hunk 1]
```diff
@@ -2488,22 +2501,14 @@ object ZIO extends ZIOFunctions {
   private val _IdentityFn: Any => Any    = (a: Any) => a
   private[zio] def identityFn[A]: A => A = _IdentityFn.asInstanceOf[A => A]
 
-  implicit final class ZIOInvariant[R, E, A](private val self: ZIO[R, E, A]) extends AnyVal {
-    final def bracket: ZIO.BracketAcquire[R, E, A] =
-      new ZIO.BracketAcquire(self)
-
-    final def bracketExit: ZIO.BracketExitAcquire[R, E, A] =
```
Comment (neko-kai) [inline core/shared/src/main/scala/zio/ZIO.scala]: forgot bracketExit!

Comment (regiskuckaertz) [reply core/shared/src/main/scala/zio/ZIO.scala]: this one is already there in fact 🧐

File: core/shared/src/main/scala/zio/ZIO.scala [hunk 2]
```diff
@@ -2488,22 +2501,14 @@ object ZIO extends ZIOFunctions {
...
-
   implicit final class ZIOAutocloseableOps[R, E, A <: AutoCloseable](private val io: ZIO[R, E, A]) extends AnyVal {
 
     /**
      * Like `bracket`, safely wraps a use and release of a resource.
      * This resource will get automatically closed, because it implements `AutoCloseable`.
      */
     def bracketAuto[R1 <: R, E1 >: E, B](use: A => ZIO[R1, E1, B]): ZIO[R1, E1, B] =
-      io.bracket(a => UIO(a.close()))(use)
+      io.bracket[R1, E1, B](a => UIO(a.close()), use)
```
Comment (neko-kai) [inline core/shared/src/main/scala/zio/ZIO.scala]: Doesn't that look like a reason for invariant syntax to be there? 🤔

Comment (jdegoes) [reply core/shared/src/main/scala/zio/ZIO.scala]: Ah, right...

Comment (jdegoes) [reply core/shared/src/main/scala/zio/ZIO.scala]: Oops. Backout?

Comment (regiskuckaertz) [reply core/shared/src/main/scala/zio/ZIO.scala]: I guess so 😟 then remove `bracketExit` from the main class too?

**Comment by jdegoes** [comment]
Probably so if we can show type inference issues...

**Comment by regiskuckaertz** [comment]
I don't think it needs to be reverted: https://github.com/zio/zio/pull/1696

**Pull request merged**